services-core/src/solution_submission/retry.rs

```diff
@@ -149,13 +140,15 @@ impl RetryWithGasPriceIncrease {
         })
     }
 
-    async fn retry_(&self, args: Args, cancel_after: impl Future) -> RetryResult {
-        log::debug!("starting retry with gas price cap {}", args.gas_price_cap);
-
-        let gas_price_stream = gas_price_increase::enforce_minimum_increase_and_cap(
-            args.gas_price_cap,
-            self.gas_price_stream(args.target_confirm_time),
-        );
+    async fn retry<TransactionSender, CancellationSender>(
+        transaction_sender: TransactionSender,
```

Comment:
This function no longer uses anything from Self anymore which makes the tests a lot simpler to understand because there is no GasPriceEstimating or AsycnSleep going on.